src/olympia/promoted/models.py

```diff
@@ -38,36 +38,30 @@ def all_for(self, addon):
     def approved_for(self, addon):
         if not addon.current_version:
             return self.none()
-        # For each PromotedGroup, the group should have an
-        # associated promoted_version version that is:
-        # 1. The addon's current version
-        # 2. The same promoted group
-        # 3. The group has approved_applications for the current version
-        # OR is not prereviewed for all apps
-        groups = (
-            self.all_for(addon=addon)
-            .filter(
-                Q(
-                    promoted_versions__version=addon.current_version,
-                    promotedaddonpromotion__promoted_group=models.F('id'),
-                )
-                | Q(
-                    promotedaddonpromotion__promoted_group__listed_pre_review=False,
-                    promotedaddonpromotion__promoted_group__unlisted_pre_review=False,
+
+        # An addon is approved for a promoted group if:
+        # 1. For each PromotedAddonVersion A, there exists a
```

Comment:
should the models be reversed in the comment? is A really PromotedAddonVersion or is it PromotedAddonPromotion?

Reply:
Argh, yeah, the other way around.